### PR TITLE
Improve documentation and checks of WENO buffers

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

I got confused by my own code from a month ago, so clearly the documentation and checking needed improvement 😛 

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [x] Code
- [x] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsvu — DISMISSED** [review]

**Review by nilsvu — COMMENTED** [review]
Seems fairly uncontroversial 😄

File: src/Evolution/DiscontinuousGalerkin/Limiters/HwenoImpl.hpp [hunk 1]
```diff
@@ -535,6 +539,9 @@ void hweno_impl(
         std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
         boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
         neighbor_data) noexcept {
+  ASSERT(modified_neighbor_solution_buffer->size() == neighbor_data.size(),
+         "Was the buffer modified_neighbor_solution_buffer initialized?");
```
Comment (carmaza) [inline src/Evolution/DiscontinuousGalerkin/Limiters/HwenoImpl.hpp]: I'd prepend "Sizes of.... don't match." to the message.

File: src/Evolution/DiscontinuousGalerkin/Limiters/SimpleWenoImpl.hpp [hunk 2]
```diff
@@ -52,6 +57,9 @@ void simple_weno_impl(
         std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
         boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
         neighbor_data) noexcept {
+  ASSERT(modified_neighbor_solution_buffer->size() == neighbor_data.size(),
+         "Was the buffer modified_neighbor_solution_buffer initialized?");
```
Comment (carmaza) [inline src/Evolution/DiscontinuousGalerkin/Limiters/SimpleWenoImpl.hpp]: same

**Review by carmaza — COMMENTED** [review]
Squash. I don't have more comments.

File: src/Evolution/DiscontinuousGalerkin/Limiters/HwenoImpl.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/Limiters/HwenoImpl.hpp]: I've always found these sorts of error messages in SpEC not all that helpful. I agree with Cris that the size mismatch should be printed, and also what the two sizes are

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by fmahebert** [comment]
@nilsdeppe @carmaza Updated the error messages

**Review by carmaza — APPROVED** [review]
The content and format of new messages look fine to me.

**Review by nilsdeppe — APPROVED** [review]
image

**Review by nilsvu — APPROVED** [review]

**Comment by fmahebert** [comment]
One of the unit tests timed out in the macOS build...

**Pull request merged**